Compiler crash with "Assertion failed: signature || !origType->hasTypeParameter()"

### Description

I get this crash building an internal app with a locally built toolchain on Windows from swift/main sync'd 5 days ago. I thought this is https://github.com/apple/swift/issues/73198 but its fix https://github.com/apple/swift/pull/73215 doesn't fix it. So splitting that into an new issue here.

### Reproduction

See the subsequent comment: https://github.com/apple/swift/issues/73345#issuecomment-2103621337

### Stack dump

```text
error: compile command failed due to exception 3 (use -v to see invocation)
Assertion failed: signature || !origType->hasTypeParameter(), file S:\SourceCache\swift\include\swift/SIL/AbstractionPattern.h, line 529
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the crash backtrace.
Stack dump:
0.      Program arguments: S:/PROGRA~1/Swift/TOOLCH~1/000611~1.0_A/usr/bin/swift-frontend.exe -frontend -c C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\BaseModel+FirebaseData.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\BatchKeyValueCollector.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\DocumentChangeHelpers.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\EncryptedEntity+FirebaseData.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\FirebaseSidebarSyncClient+Impl.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\FirebaseSidebarSyncClient+Model.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\FirebaseSidebarSyncClient.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\FirebaseSidebarSyncTestTool.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\Logger.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\PublishTransaction.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\SyncMode+Combine.swift C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\TimeThrottlingConfig.swift -primary-file C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\TimeThrottlingRebatcher.swift -emit-reference-dependencies-path C:\\Users\\hiroshi\\arc\\build\\debug\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\CMakeFiles\\FirebaseSidebarSyncClient.dir\\TimeThrottlingRebatcher.swift.obj.swiftdeps -target x86_64-unknown-windows-msvc -disable-objc-interop -cxx-interoperability-mode=default -sdk "S:\\Program Files\\Swift\\Platforms\\Windows.platform\\Developer\\SDKs\\Windows.sdk" -windows-sdk-version 10.0.22000.0 -I C:\\Users\\hiroshi\\arc\\build\\debug\\swift -I C:\\Users\\hiroshi\\arc\\build\\debug\\_deps\\opencombine-src\\Sources\\COpenCombineHelpers\\include -I C:\\Users\\hiroshi\\arc\\build\\debug\\_deps\\swiftfirebase-src\\Sources\\firebase\\include -I C:\\Users\\hiroshi\\arc\\build\\NugetPackages\\com.google.firebase.windows.amd64.0.0.0-f6d6ee2\\include -I C:\\Users\\hiroshi\\arc\\Frameworks\\ARCClients\\Sources\\LevelDBChecker\\Headers -I C:\\Users\\hiroshi\\arc\\build\\debug\\generated\\Frameworks\\WinRT\\Sources\\CWinRT\\include -I "S:\\Program Files\\Swift\\Toolchains\\0.0.0+Asserts\\usr\\include" -I C:\\Users\\hiroshi\\arc\\build\\NugetPackages\\org.libsodium.libsodium.windows.amd64.0.0.0-ec57a46\\include -enable-testing -module-link-name FirebaseSidebarSyncClient -static -Onone -D SWIFT_PACKAGE -D DEBUG -g -debug-info-format=dwarf -disable-round-trip-debug-types -disable-large-loadable-types-reg2mem -empty-abi-descriptor -Xcc -working-directory -Xcc C:\\Users\\hiroshi\\arc\\build\\debug -resource-dir S:\\PROGRA~1\\Swift\\TOOLCH~1\\000611~1.0_A\\usr\\lib\\swift -Xcc -DINTERNAL_EXPERIMENTAL -Xcc -DSODIUM_STATIC -Xcc -DINTERNAL_EXPERIMENTAL -Xcc -DSR69711 -module-name FirebaseSidebarSyncClient -plugin-path S:\\PROGRA~1\\Swift\\TOOLCH~1\\000611~1.0_A\\usr\\bin -plugin-path S:\\PROGRA~1\\Swift\\TOOLCH~1\\000611~1.0_A\\usr\\local\\bin -parse-as-library -num-threads 64 -o C:\\Users\\hiroshi\\arc\\build\\debug\\Frameworks\\ARCClients\\Sources\\FirebaseSidebarSyncClient\\CMakeFiles\\FirebaseSidebarSyncClient.dir\\TimeThrottlingRebatcher.swift.obj -index-store-path C:\\Users\\hiroshi\\arc\\build\\debug\\index -index-system-modules
1.      Swift version 6.0-dev (LLVM 384be736a3af7f0, Swift 3d32f5fc8a9087a)
2.      Compiling with effective version 5.10
3.      While evaluating request ASTLoweringRequest(Lowering AST to SIL for file "C:\Users\hiroshi\arc\Frameworks\ARCClients\Sources\FirebaseSidebarSyncClient\TimeThrottlingRebatcher.swift")
4.      While silgen emitFunction SIL function "@$s25FirebaseSidebarSyncClient23TimeThrottlingRebatcherC13planNextEvent33_511274859BA13DF783665B0F348B6BACLL2inySd_tF".
 for 'planNextEvent(in:)' (at C:\Users\hiroshi\arc\Frameworks\ARCClients\Sources\FirebaseSidebarSyncClient\TimeThrottlingRebatcher.swift:119:13)
Exception Code: 0x80000003
 #0 0x00007ff7a5e07485 HandleAbort (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x73a7485)
 #1 0x00007ff9a5f4e695 (C:\WINDOWS\System32\ucrtbase.dll+0x7e695)
 #2 0x00007ff9a5f4f6a1 (C:\WINDOWS\System32\ucrtbase.dll+0x7f6a1)
 #3 0x00007ff9a5f5104e (C:\WINDOWS\System32\ucrtbase.dll+0x8104e)
 #4 0x00007ff9a5f51291 (C:\WINDOWS\System32\ucrtbase.dll+0x81291)
 #5 0x00007ff79f387e29 swift::Lowering::AbstractionPattern::initSwiftType(class swift::SubstitutionMap, class swift::CanGenericSignature, class swift::CanType, enum swift::Lowering::AbstractionPattern::Kind) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x927e29)
 #6 0x00007ff7a022c3f5 swift::SILFunctionConventions::isTypedError(void) const (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x17cc3f5)
 #7 0x00007ff7a0224f0e swift::SILResultInfo::getReturnValueType(class swift::SILModule &, class swift::SILFunctionType const *, class swift::TypeExpansionContext) const (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x17c4f0e)
 #8 0x00007ff7a0211748 clang::QualType::operator->(void) const (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x17b1748)
 #9 0x00007ff7a0222f42 swift::Lowering::TypeConverter::getNSStringType(void) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x17c2f42)
#10 0x00007ff7a022750a swift::SILFunction::getTypeExpansionContext(void) const (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x17c750a)
#11 0x00007ff7a021d063 swift::Lowering::TypeConverter::getConstantInfo(class swift::TypeExpansionContext, struct swift::SILDeclRef) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x17bd063)
#12 0x00007ff79f775b7f BoxInitialization::finishInitialization(class swift::Lowering::SILGenFunction &) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xd15b7f)
#13 0x00007ff79f776b0b BoxInitialization::getAddressForInPlaceInitialization(class swift::Lowering::SILGenFunction &, class swift::SILLocation) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xd16b0b)
#14 0x00007ff79f771cd4 swift::Lowering::SILGenFunction::emitSetAccessor(class swift::SILLocation, struct swift::SILDeclRef, class swift::SubstitutionMap, class swift::Lowering::ArgumentSource &&, bool, bool, class swift::Lowering::PreparedArguments &&, class swift::Lowering::ArgumentSource &&, bool) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xd11cd4)
#15 0x00007ff79f76b211 swift::Lowering::SILGenFunction::emitGetAccessor(class swift::SILLocation, struct swift::SILDeclRef, class swift::SubstitutionMap, class swift::Lowering::ArgumentSource &&, bool, bool, class swift::Lowering::PreparedArguments &&, class swift::Lowering::SGFContext, bool, class std::optional<class swift::ActorIsolation>) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xd0b211)
#16 0x00007ff79f73d8d4 swift::Lowering::LValue::forValue(enum swift::Lowering::SGFAccessKind, class swift::Lowering::ManagedValue, class swift::CanType) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcdd8d4)
#17 0x00007ff79f7357ce swift::Lowering::SILGenFunction::emitLoadOfLValue(class swift::SILLocation, class swift::Lowering::LValue &&, class swift::Lowering::SGFContext, bool) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcd57ce)
#18 0x00007ff79f736d49 swift::Lowering::SILGenFunction::emitRValueForNonMemberVarDecl(class swift::SILLocation, class swift::ConcreteDeclRef, class swift::CanType, enum swift::AccessSemantics, class swift::Lowering::SGFContext) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcd6d49)
#19 0x00007ff79f698393 swift::Lowering::SILGenFunction::emitRValueForDecl(class swift::SILLocation, class swift::ConcreteDeclRef, class swift::Type, enum swift::AccessSemantics, class swift::Lowering::SGFContext) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc38393)
#20 0x00007ff79f6a2f49 llvm::SmallVectorTemplateBase<class swift::Lowering::RValue, 0>::takeAllocationForGrow(class swift::Lowering::RValue *, unsigned __int64) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc42f49)
#21 0x00007ff79f698179 swift::Lowering::SILGenFunction::emitRValueAsSingleValue(class swift::Expr *, class swift::Lowering::SGFContext) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc38179)
#22 0x00007ff79f76bbb8 swift::Lowering::SILGenFunction::emitGetAccessor(class swift::SILLocation, struct swift::SILDeclRef, class swift::SubstitutionMap, class swift::Lowering::ArgumentSource &&, bool, bool, class swift::Lowering::PreparedArguments &&, class swift::Lowering::SGFContext, bool, class std::optional<class swift::ActorIsolation>) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xd0bbb8)
#23 0x00007ff79f75c0cb llvm::CastInfo<class swift::OptionalEvaluationExpr, class swift::Expr *, void>::doCastIfPossible(class swift::Expr *const &) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcfc0cb)
#24 0x00007ff79f76e7e3 swift::Lowering::SILGenFunction::emitMonomorphicApply(class swift::SILLocation, class swift::Lowering::ManagedValue, class llvm::ArrayRef<class swift::Lowering::ManagedValue>, class swift::CanType, class swift::CanType, class swift::optionset::OptionSet<enum swift::ApplyFlags, unsigned char>, class std::optional<enum swift::SILFunctionTypeRepresentation>, class std::optional<class swift::ForeignErrorConvention> const &, class swift::Lowering::SGFContext) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xd0e7e3)
#25 0x00007ff79f75c2df llvm::CastInfo<class swift::OptionalEvaluationExpr, class swift::Expr *, void>::doCastIfPossible(class swift::Expr *const &) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcfc2df)
#26 0x00007ff79f762149 swift::Lowering::SILGenFunction::emitApplyWithRethrow(class swift::SILLocation, class swift::SILValue, class swift::SILType, class swift::SubstitutionMap, class llvm::ArrayRef<class swift::SILValue>) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xd02149)
#27 0x00007ff79f755fcd swift::Lowering::PreparedArguments::add(class swift::SILLocation, class swift::Lowering::RValue &&) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcf5fcd)
#28 0x00007ff79f7547e9 swift::Lowering::PreparedArguments::add(class swift::SILLocation, class swift::Lowering::RValue &&) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcf47e9)
#29 0x00007ff79f75f970 swift::Lowering::SILGenFunction::emitApplyExpr(class swift::ApplyExpr *, class swift::Lowering::SGFContext) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcff970)
#30 0x00007ff79f6a3c9f llvm::SmallVectorTemplateBase<class swift::Lowering::RValue, 0>::takeAllocationForGrow(class swift::Lowering::RValue *, unsigned __int64) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc43c9f)
#31 0x00007ff79f6980d1 swift::Lowering::SILGenFunction::emitRValue(class swift::Expr *, class swift::Lowering::SGFContext) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc380d1)
#32 0x00007ff79f69262f swift::Lowering::SILGenFunction::emitIgnoredExpr(class swift::Expr *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc3262f)
#33 0x00007ff79f716484 swift::Lowering::SILGenFunction::prepareIndirectResultInit(class swift::SILLocation, class swift::Lowering::AbstractionPattern, class swift::CanType, class llvm::SmallVectorImpl<class swift::SILValue> &, class llvm::SmallVectorImpl<class swift::DiverseStackBase::stable_iterator> &) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcb6484)
#34 0x00007ff79f71541f swift::Lowering::SILGenFunction::prepareIndirectResultInit(class swift::SILLocation, class swift::Lowering::AbstractionPattern, class swift::CanType, class llvm::SmallVectorImpl<class swift::SILValue> &, class llvm::SmallVectorImpl<class swift::DiverseStackBase::stable_iterator> &) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcb541f)
#35 0x00007ff79f712813 swift::Lowering::SILGenFunction::emitStmt(class swift::Stmt *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xcb2813)
#36 0x00007ff79f6bfab6 swift::Lowering::SILGenFunction::emitFunction(class swift::FuncDecl *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc5fab6)
#37 0x00007ff79f62233b swift::Lowering::SILGenModule::emitFunctionDefinition(struct swift::SILDeclRef, class swift::SILFunction *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbc233b)
#38 0x00007ff79f6254c9 swift::Lowering::SILGenModule::emitOrDelayFunction(struct swift::SILDeclRef) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbc54c9)
#39 0x00007ff79f621575 swift::Lowering::SILGenModule::emitFunction(class swift::FuncDecl *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbc1575)
#40 0x00007ff79f672338 SILGenExtension::visitFuncDecl(class swift::FuncDecl *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc12338)
#41 0x00007ff79f66c6dd swift::Lowering::SILGenModule::emitSelfConformanceWitnessTable(class swift::ProtocolDecl *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc0c6dd)
#42 0x00007ff79f672758 swift::Lowering::SILGenModule::visitNominalTypeDecl(class swift::NominalTypeDecl *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xc12758)
#43 0x00007ff79f625767 swift::Lowering::SILGenModule::emitPropertyWrapperBackingInitializer(class swift::VarDecl *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbc5767)
#44 0x00007ff79f625ef4 swift::ASTLoweringRequest::evaluate(class swift::Evaluator &, struct swift::ASTLoweringDescriptor) const (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbc5ef4)
#45 0x00007ff79f630bd6 swift::SimpleRequest<class swift::ASTLoweringRequest, (struct swift::ASTLoweringDescriptor), 9>::evaluateRequest(class swift::ASTLoweringRequest const &, class swift::Evaluator &) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbd0bd6)
#46 0x00007ff79f61b632 swift::Evaluator::getResultUncached<class swift::ASTLoweringRequest, class `class std::unique_ptr<class swift::SILModule, struct std::default_delete<class swift::SILModule>> __cdecl swift::evaluateOrFatal<class swift::ASTLoweringRequest>(class swift::Evaluator &, class swift::ASTLoweringRequest)'::`2'::<lambda_1>>(class swift::ASTLoweringRequest const &, class `class std::unique_ptr<class swift::SILModule, struct std::default_delete<class swift::SILModule>> __cdecl swift::evaluateOrFatal<class swift::ASTLoweringRequest>(class swift::Evaluator &, class swift::ASTLoweringRequest)'::`2'::<lambda_1>) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbbb632)
#47 0x00007ff79f62f1f1 swift::performASTLowering(class swift::FileUnit &, class swift::Lowering::TypeConverter &, class swift::SILOptions const &, class swift::IRGenOptions const *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0xbcf1f1)
#48 0x00007ff79edf146c swift::performCompileStepsPostSema(class swift::CompilerInstance &, int &, class swift::FrontendObserver *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x39146c)
#49 0x00007ff79edefd08 std::basic_stringbuf<char, struct std::char_traits<char>, class std::allocator<char>>::pbackfail(int) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x38fd08)
#50 0x00007ff79edf027e std::basic_stringbuf<char, struct std::char_traits<char>, class std::allocator<char>>::pbackfail(int) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x39027e)
#51 0x00007ff79edf27db swift::performFrontend(class llvm::ArrayRef<char const *>, char const *, void *, class swift::FrontendObserver *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x3927db)
#52 0x00007ff79ec30b3b llvm::SmallVectorTemplateBase<char const *, 1>::push_back(char const *) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x1d0b3b)
#53 0x00007ff79ec306dc swift::mainEntry(int, char const **) (S:\PROGRA~1\Swift\TOOLCH~1\000611~1.0_A\usr\bin\swift-frontend.exe+0x1d06dc)
#54 0x00007ff7a5e7e148 invoke_main D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78:0
#55 0x00007ff7a5e7e148 __scrt_common_main_seh D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288:0
#56 0x00007ff9a6c1257d (C:\WINDOWS\System32\KERNEL32.DLL+0x1257d)
#57 0x00007ff9a8c0aa48 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x5aa48)
ninja: build stopped: subcommand failed.
```


### Expected behavior

No crash

### Environment

Windows

### Additional information

_No response_

--- a/lib/SIL/IR/TypeLowering.cpp
+++ b/lib/SIL/IR/TypeLowering.cpp
@@ -4441,6 +4441,12 @@ TypeConverter::getLoweredLocalCaptures(SILDeclRef fn) {
           if (auto *var = actorIsolation.getActorInstance()) {
             assert(isa<ParamDecl>(var));
             recordCapture(CapturedValue(var, 0, afd->getLoc()));
+	    if (var->getInterfaceType()->hasTypeParameter()) {
+	      // If the isolated parameter is of a generic (actor)
+	      // type, we need to treat as if the local function is
+	      // generic.
+	      capturesGenericParams = true;
+	    }
           }
         }
       }

--- a/test/SILGen/local_function_isolation.swift
+++ b/test/SILGen/local_function_isolation.swift
@@ -53,3 +53,17 @@ func f(isolation: isolated MyActor, ns: NotSendable) {
 }
 
 func test() async {}
+
+// A generic actor type, which causes the generic self parameter
+// (actor isolation parameter) to be added to captures of the
+// local/nested function.
+actor GenericActor<K> {
+  var i: Int = 0
+  private func outerFunc() {
+    func accessSelf() -> Int {
+      // CHECK-LABEL: sil private [ossa] @$s24local_function_isolation12GenericActorC9outerFunc33_7B9E2B75110B8600A136A469D51CAF2BLLyyF10accessSelfL_SiylF : $@convention(thin) <K> (@sil_isolated @guaranteed GenericActor<K>) -> Int {
+      return 0
+    }
+    print(accessSelf())
+  }
+}

